app/components/search-facet-source/component.js

```diff
@@ -0,0 +1,69 @@
+import { computed } from '@ember/object';
+
+import TypeaheadComponent from '../search-facet-typeahead/component';
+
+
+export default TypeaheadComponent.extend({
+    donut: null,
+
+    sourcesList: computed('aggregations', function() {
+        const data = this.get('aggregations.sources.buckets');
+        return data ? data.mapBy('key') : [];
+    }),
+
+    didReceiveAttrs() {
+        if (!this.get('loading') && this.get('aggregations')) {
+            const data = this.get('aggregations.sources.buckets');
+            this.updateDonut(data);
+        }
+    },
+
+    updateDonut(data) {
+        // eslint-disable-next-line camelcase
+        const columns = data.map(({ key, doc_count }) => [key, doc_count]);
+        const title = columns.length + (columns.length === 1 ? ' Source' : ' Sources');
+
+        const donut = this.get('donut');
+        if (!donut) {
+            this.initDonut(title, columns);
+        } else {
+            donut.load({
+                columns,
+                unload: true,
+            });
+            this.$('.c3-chart-arcs-title').text(title);
+        }
+    },
+
+    initDonut(title, columns) {
+        try {
+            const element = this.$('.donut').get(0);
+            // eslint-disable-next-line no-undef
+            const donut = c3.generate({
+                bindto: element,
+                data: {
+                    columns,
+                    type: 'donut',
+                    onclick: (d) => {
+                        const selected = this.get('selected');
+                        if (!selected.includes(d.name)) {
+                            this.send('changeFilter', [d.name, ...selected]);
+                        }
+                    },
+                },
+                legend: { show: false },
+                donut: {
+                    title,
+                    label: {
+                        show: false,
+                    },
+                },
+                size: { height: 200 },
+            });
+            this.set('donut', donut);
+        } catch (e) {
+            // eslint-disable-next-line no-console
+            console.debug(e);
```

Comment:
Not sure if this should be debug or error

Reply:
Wasn't sure the exact difference between all of em, so found the below. Doesn't really matter. If we are concerned about it, a proper error to sentry would be better,  but I don't really care.

```
Technically console.log console.debug and console.info are identical
However the way they display the data is little different

console.log Black color text with no icon

console.info Blue color text with icon

console.debug Pure black color text

console.warn Yellow color text with icon

console.error Red Color text with icon
```
https://stackoverflow.com/questions/21876461/difference-between-console-log-and-console-debug

Reply:
Are sentry errors only sent for uncaught exceptions or is there a way to send them directly?

Reply:
@fabmiz, want to comment here? You've messed with sending things to sentry more than I have.

Reply:
Looks like it's definitely possible https://docs.sentry.io/clients/javascript/usage/#reporting-errors-correctly